src/lib.rs

```diff
@@ -1054,6 +1070,55 @@ where
     }
 }
 
+// borsh is byte-size little-endian de-needs-external-schema no-bit-compression serde
+// current ser/de for it is not optimal impl because const math is not stable nor primitives has bits traits
+#[cfg(feature = "borsh")]
+impl<T: BorshSerialize, const BITS: usize> BorshSerialize for UInt<T, BITS>
+where
+    Self: Number,
+{
+    fn serialize<W: borsh::io::Write>(&self, writer: &mut W) -> borsh::io::Result<()> {
+        self.value.serialize(writer)
```

Comment:
If we later improve this to be more space efficient, would we be breaking anything? Do we have to guarantee binary compatibility with earlier/later versions?